src/test_c2s_srv.c

```diff
@@ -49,30 +49,32 @@
  * @param spds[] [] speed check array
  * @param spd_index  index used for speed check array
  * @param conn_options Connection options
+ * @param ctx The SSL context (possibly NULL)
  * @return 0 - success,
  *          >0 - error code
  *          Error codes:
  *          -1 - Listener socket creation failed
  *          -100 - timeout while waiting for client to connect to server�s ephemeral port
- * 			-errno - Other specific socket error numbers
- *			-101 - Retries exceeded while waiting for client to connect
- *			-102 - Retries exceeded while waiting for data from connected client
+ *                         -errno - Other specific socket error numbers
+ *                        -101 - Retries exceeded while waiting for client to connect
+ *                        -102 - Retries exceeded while waiting for data from connected client
  *
  */
-int test_c2s(int ctlsockfd, tcp_stat_agent* agent, TestOptions* testOptions,
+int test_c2s(Connection* ctl, tcp_stat_agent* agent, TestOptions* testOptions,
              int conn_options, double* c2sspd, int set_buff, int window,
              int autotune, char* device, Options* options, int record_reverse,
-             int count_vars, char spds[4][256], int* spd_index) {
+             int count_vars, char spds[4][256], int* spd_index, SSL_CTX* ctx) {
```

Comment:
It seems strange that this should be independent of the `Connection` struct. Could this be part of that struct?